### PR TITLE
Improving Transform Tests

### PR DESCRIPTION
**Pull request by saeeedtorabi**
- improved tests
- added new tests (test coverage is now over 99%)
- fixed minor bugs
- improved formatting and doc style of different classes
- addressed Dave comments

### CODE REVIEW

File: core/geometry/src/geometry3d/Transform.ts [hunk 1]
```diff
@@ -391,49 +397,54 @@ export class Transform implements BeJSONFunctions {
...
+   * * If `p = (x,y,z)` then this method computes `Tp = M*p + o` and returns it as 3 elements of the array.
    */
   public multiplyXYZToFloat64Array(x: number, y: number, z: number, result?: Float64Array): Float64Array {
     return Matrix3d.xyzPlusMatrixTimesCoordinatesToFloat64Array(this._origin, this._matrix, x, y, z, result);
   }
   /**
-   * Treat the 3x3 `matrix` and `origin` as upper 3x4 part of a 4x4 matrix, with 0001 as the final row. Now multiply
-   * the transposed of this 4x4 matrix by Point4d given as xyzw. Return as a new point4d (`M*p` as first 3 elements
-   * and `o*p + w` as last element where `p = (x,y,z)`) or in the pre-allocated result (if result is given).
+   * Multiply the homogeneous point by the transpose if `this` Transform. Return as a new `Point4d` or in the
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Transform.ts:442]:
```suggestion
   * Multiply the homogeneous point by the transpose of `this` Transform. Return as a new `Point4d` or in the
```

File: core/geometry/src/geometry3d/Transform.ts [hunk 2]
```diff
@@ -391,49 +397,54 @@ export class Transform implements BeJSONFunctions {
...
+   * * Logically, this is multiplication by the 4x4 matrix formed from the 3x4 instance augmented with fourth row 0001.
    */
   public multiplyXYZWToFloat64Array(x: number, y: number, z: number, w: number, result?: Float64Array): Float64Array {
     return Matrix3d.xyzPlusMatrixTimesWeightedCoordinatesToFloat64Array(this._origin, this._matrix, x, y, z, w, result);
   }
   /**
-   * If `p = (x,y,z)` then transform is `Tp = M*p + o`. This function returns the transformed point as a new
-   * Float64Array with size 3 (`Tp` as 3 elements) or in the pre-allocated result (if result is given).
+   * * Transform the point. Return as new `Float64Array` with size 3, or in the pre-allocated result (if result is given).
+   * * If `p = (x,y,z)` then this method computes `Tp = M*p + o` and returns it as 3 elements of the array.
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Transform.ts:436]:
```suggestion
   * * If `p = (x,y,z)` then this method computes `Tp = M*p + o` and returns it as the first 3 elements of the array.
```

File: core/geometry/src/geometry3d/Transform.ts [hunk 3]
```diff
@@ -336,20 +341,21 @@ export class Transform implements BeJSONFunctions {
     return Transform.createRefs(undefined, matrix.clone());
   }
   /**
-   * Create a transform with the specified `matrix` and points `a` and `b`. The returned Transform, transforms
-   * point `p` to `M*(p-a) + b` (i.e., `Tp = M*(p-a) + b`) so transforms point `a` to 'b'.
+   * Create a transform with the specified `matrix` and points `a` and `b`. The returned Transform, maps
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Transform.ts:344]:
```suggestion
   * Create a transform with the specified `matrix` and points `a` and `b`. The returned Transform maps
```

File: core/geometry/src/geometry3d/Transform.ts [hunk 4]
```diff
@@ -336,20 +341,21 @@ export class Transform implements BeJSONFunctions {
...
   ): Transform {
-    // we define the origin o = b - M*a so Tp = M*p + o = M*p + (b - M*a) = M*(x-a) + b
+    // we define the origin o = b - M*a so Tp = M*p + o = M*p + (b - M*a) = M*(p-a) + b
     const origin = Matrix3d.xyzMinusMatrixTimesXYZ(b, matrix, a);
     return Transform.createRefs(origin, matrix.clone(), result);
   }
   /**
    * Create a Transform which leaves the fixedPoint unchanged and scales everything else around it by
-   * a single scale factor. The returned Transform, transforms a point `p` to `M*p + (f - M*f)`
+   * a single scale factor. The returned Transform, maps a point `p` to `M*p + (f - M*f)`
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Transform.ts:356]:
```suggestion
   * a single scale factor. The returned Transform maps a point `p` to `M*p + (f - M*f)`
```

File: core/geometry/src/geometry3d/Transform.ts [hunk 5]
```diff
@@ -391,49 +397,54 @@ export class Transform implements BeJSONFunctions {
     return Matrix3d.xyzPlusMatrixTimesCoordinates(this._origin, this._matrix, x, y, z, result);
   }
   /**
-   * Multiply a specific row (component) of the transform matrix times xyz and add it to the origin element
-   * at the same row. Return the result.
+   * Multiply a specific row (component) of the transform 3x4 instance times (x,y,z,1). Return the result.
```
Comment (dassaf4) [inline core/geometry/src/geometry3d/Transform.ts:400]:
```suggestion
   * Multiply a specific row (component) of the 3x4 instance times (x,y,z,1). Return the result.
```

**Review by dassaf4 — APPROVED** [review]

File: core/geometry/src/geometry3d/Transform.ts (hunk 1 above)
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Transform.ts:442]: will fix

File: core/geometry/src/geometry3d/Transform.ts (hunk 3 above)
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Transform.ts:344]: will fix

File: core/geometry/src/geometry3d/Transform.ts (hunk 4 above)
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Transform.ts:356]: will fix

File: core/geometry/src/geometry3d/Transform.ts (hunk 5 above)
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Transform.ts:400]: will fix

File: core/geometry/src/geometry3d/Transform.ts [hunk 6]
```diff
@@ -391,49 +397,54 @@ export class Transform implements BeJSONFunctions {
...
    */
   public multiplyComponentXYZ(componentIndex: number, x: number, y: number, z: number = 0): number {
     const coffs = this._matrix.coffs;
     const idx = 3 * componentIndex;
     return this.origin.at(componentIndex) + (coffs[idx] * x) + (coffs[idx + 1] * y) + (coffs[idx + 2] * z);
   }
   /**
-   * Multiply a specific row (component) of the transform matrix times xyz and add it to the origin element
-   * at the same row times w. Return the result.
+   * Multiply a specific row (component) of the transform 3x4 instance times (x,y,z,w). Return the result.
```
[the remark this answers is not in the document]
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Transform.ts:408]: will fix

File: core/geometry/src/geometry3d/Transform.ts (hunk 2 above)
Comment (saeeedtorabi) [reply core/geometry/src/geometry3d/Transform.ts:436]: will fix

**Pull request merged**